docs/how-to/pin-files.md

```diff
@@ -12,13 +11,46 @@ While this is nice, sometimes you want to be able to control what you keep aroun
 
 Let's look at this example to explore pinning to your local IPFS node in a bit more depth:
 
-```
-echo "ipfs rocks" > foo
-ipfs add foo
-ipfs pin ls --type=all
-ipfs pin rm <foo hash>
-ipfs pin rm <foo hash>
-ipfs pin ls --type=all
+1. First, create a file called `foo` with the contents `ipfs rocks`:
+
+    ```shell
+    echo "ipfs rocks" > foo    
+    ```
+
+    The `echo` command does not give an output when using the _output redirection_ `>` instruction.
+
+1. Next, add `foo` to IPFS using the `ipfs add` command:
+
+    ```shell
+    ipfs add foo               
+
+    > added QmRTV3h1jLcACW4FRfdisokkQAk4E4qDhUzGpgdrd4JAFy foo
+    ```
+
+
+ipfs pin ls --type=all     
+
+> QmQPeNsJPyVWPFDVHb77w8G42Fvo15z4bG2X8D2GhfbSXc recursive
+> QmRTV3h1jLcACW4FRfdisokkQAk4E4qDhUzGpgdrd4JAFy recursive
+> QmQy6xmJhrcC5QLboAcGFcAE1tC8CrwDVkrHdEYJkLscrQ indirect
+> ...
+
+
+ipfs pin rm <foo hash>     
+
+> unpinned QmRTV3h1jLcACW4FRfdisokkQAk4E4qDhUzGpgdrd4JAFy
+
+
+ipfs pin rm <foo hash>     
+
+> Error: not pinned or pinned indirectly
+
+
+ipfs pin ls --type=all    
+
+> QmQPeNsJPyVWPFDVHb77w8G42Fvo15z4bG2X8D2GhfbSXc recursive
+> QmQy6xmJhrcC5QLboAcGFcAE1tC8CrwDVkrHdEYJkLscrQ indirect
+> ...
```

Comment:
@TMoMoreau following the structure I've laid out above, can you try and explain what each of these commands does?